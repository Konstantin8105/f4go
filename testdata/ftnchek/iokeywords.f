C  Exerciser for recognition and flagging of various I/O keywords.
      program iokeywds
      call vms_isms
      call direct_access
      call f90_stuff
      end
      subroutine vms_isms
      character*512 line(4)
C  This one uses some VMS-isms
      open(unit=1,carriagecontrol='none',recordtype='fixed',
     $  recl=2048,status='new',name='file1')
      open(unit=2,carriagecontrol='none',recl=512,status='old',err=1)
      goto 2
1     stop 'Please $ASSIGN/USER output FOR001'
2     do i=1,4
       read(2,'(a)',err=3) line(i)
      end do
      write(1,'(a)') line(1)//line(2)//line(3)//line(4)
      goto 2
3     continue
      close(unit=1)
      close(unit=2)
      end
      subroutine direct_access
      character inline*79

      open(unit=1,file='test',access='direct',form='formatted',
     1     status='new', recl=80,err=909)
*
*       Read a line from standard input, write it to the new file,
*       and echo it to standard output.
*
      do j = 1,100
            read(5,10,end=910)inline
            write(1,20,rec=j)inline
            write(5,20)inline
      enddo

 10   format(79a)
 20   format(1x79a)
      GoTo 910
 909  print*,'Error opening file.'
 910  continue
      close(unit=1)
      close(unit=5)
      end
      subroutine f90_stuff
      integer unit,inputsize,val,i1
      character *20 c1,c2
      unit = 3
      open(unit=unit,file='data.dat',status='old',action='read',
     $     access='sequential',form='formatted',err=910)
      read(unit=unit,fmt='(i10)',advance='no',size=inputsize,
     $     err=920) val
      inquire(unit=unit,recl=i1,access=c1,position=c2)
      write(unit=*,fmt='(1x,a7)',advance='yes') 'Success'
      close(unit)
      return
 910  print *,'Error opening file.'
      return
 920  print *,'Error reading file.'
      return
      end
