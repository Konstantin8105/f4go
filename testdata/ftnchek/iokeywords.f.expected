COMMENT             	C  Exerciser for recognition and flagging of various I/O keywords.
NEW_LINE            	

PROGRAM             	program
IDENT               	iokeywds
NEW_LINE            	

CALL                	call
IDENT               	vms_isms
NEW_LINE            	

CALL                	call
IDENT               	direct_access
NEW_LINE            	

CALL                	call
IDENT               	f90_stuff
NEW_LINE            	

END                 	end
NEW_LINE            	

SUBROUTINE          	subroutine
IDENT               	vms_isms
NEW_LINE            	

IDENT               	character
*                   	*
INT                 	512
IDENT               	line
(                   	(
INT                 	4
)                   	)
NEW_LINE            	

COMMENT             	C  This one uses some VMS-isms
NEW_LINE            	

IDENT               	open
(                   	(
IDENT               	unit
=                   	=
INT                 	1
,                   	,
IDENT               	carriagecontrol
=                   	=
STRING              	'none'
,                   	,
IDENT               	recordtype
=                   	=
STRING              	'fixed'
,                   	,
NEW_LINE            	

IDENT               	recl
=                   	=
INT                 	2048
,                   	,
IDENT               	status
=                   	=
STRING              	'new'
,                   	,
IDENT               	name
=                   	=
STRING              	'file1'
)                   	)
NEW_LINE            	

IDENT               	open
(                   	(
IDENT               	unit
=                   	=
INT                 	2
,                   	,
IDENT               	carriagecontrol
=                   	=
STRING              	'none'
,                   	,
IDENT               	recl
=                   	=
INT                 	512
,                   	,
IDENT               	status
=                   	=
STRING              	'old'
,                   	,
IDENT               	err
=                   	=
INT                 	1
)                   	)
NEW_LINE            	

IDENT               	goto
INT                 	2
NEW_LINE            	

INT                 	1
IDENT               	stop
STRING              	'Please $ASSIGN/USER output FOR001'
NEW_LINE            	

INT                 	2
DO                  	do
IDENT               	i
=                   	=
INT                 	1
,                   	,
INT                 	4
NEW_LINE            	

IDENT               	read
(                   	(
INT                 	2
,                   	,
STRING              	'(a)'
,                   	,
IDENT               	err
=                   	=
INT                 	3
)                   	)
IDENT               	line
(                   	(
IDENT               	i
)                   	)
NEW_LINE            	

END                 	end
DO                  	do
NEW_LINE            	

IDENT               	write
(                   	(
INT                 	1
,                   	,
STRING              	'(a)'
)                   	)
IDENT               	line
(                   	(
INT                 	1
)                   	)
/                   	/
/                   	/
IDENT               	line
(                   	(
INT                 	2
)                   	)
/                   	/
/                   	/
IDENT               	line
(                   	(
INT                 	3
)                   	)
/                   	/
/                   	/
IDENT               	line
(                   	(
INT                 	4
)                   	)
NEW_LINE            	

IDENT               	goto
INT                 	2
NEW_LINE            	

INT                 	3
continue            	continue
NEW_LINE            	

IDENT               	close
(                   	(
IDENT               	unit
=                   	=
INT                 	1
)                   	)
NEW_LINE            	

IDENT               	close
(                   	(
IDENT               	unit
=                   	=
INT                 	2
)                   	)
NEW_LINE            	

END                 	end
NEW_LINE            	

SUBROUTINE          	subroutine
IDENT               	direct_access
NEW_LINE            	

IDENT               	character
IDENT               	inline
*                   	*
INT                 	79
NEW_LINE            	

NEW_LINE            	

IDENT               	open
(                   	(
IDENT               	unit
=                   	=
INT                 	1
,                   	,
IDENT               	file
=                   	=
STRING              	'test'
,                   	,
IDENT               	access
=                   	=
STRING              	'direct'
,                   	,
IDENT               	form
=                   	=
STRING              	'formatted'
,                   	,
NEW_LINE            	

INT                 	1
IDENT               	status
=                   	=
STRING              	'new'
,                   	,
IDENT               	recl
=                   	=
INT                 	80
,                   	,
IDENT               	err
=                   	=
INT                 	909
)                   	)
NEW_LINE            	

COMMENT             	*
NEW_LINE            	

COMMENT             	*       Read a line from standard input, write it to the new file,
NEW_LINE            	

COMMENT             	*       and echo it to standard output.
NEW_LINE            	

COMMENT             	*
NEW_LINE            	

DO                  	do
IDENT               	j
=                   	=
INT                 	1
,                   	,
INT                 	100
NEW_LINE            	

IDENT               	read
(                   	(
INT                 	5
,                   	,
INT                 	10
,                   	,
END                 	end
=                   	=
INT                 	910
)                   	)
IDENT               	inline
NEW_LINE            	

IDENT               	write
(                   	(
INT                 	1
,                   	,
INT                 	20
,                   	,
IDENT               	rec
=                   	=
IDENT               	j
)                   	)
IDENT               	inline
NEW_LINE            	

IDENT               	write
(                   	(
INT                 	5
,                   	,
INT                 	20
)                   	)
IDENT               	inline
NEW_LINE            	

ENDDO               	enddo
NEW_LINE            	

NEW_LINE            	

INT                 	10
IDENT               	format
(                   	(
INT                 	79
IDENT               	a
)                   	)
NEW_LINE            	

INT                 	20
IDENT               	format
(                   	(
INT                 	1
IDENT               	x79a
)                   	)
NEW_LINE            	

IDENT               	GoTo
INT                 	910
NEW_LINE            	

INT                 	909
IDENT               	print
*                   	*
,                   	,
STRING              	'Error opening file.'
NEW_LINE            	

INT                 	910
continue            	continue
NEW_LINE            	

IDENT               	close
(                   	(
IDENT               	unit
=                   	=
INT                 	1
)                   	)
NEW_LINE            	

IDENT               	close
(                   	(
IDENT               	unit
=                   	=
INT                 	5
)                   	)
NEW_LINE            	

END                 	end
NEW_LINE            	

SUBROUTINE          	subroutine
IDENT               	f90_stuff
NEW_LINE            	

INTEGER             	integer
IDENT               	unit
,                   	,
IDENT               	inputsize
,                   	,
IDENT               	val
,                   	,
IDENT               	i1
NEW_LINE            	

IDENT               	character
*                   	*
INT                 	20
IDENT               	c1
,                   	,
IDENT               	c2
NEW_LINE            	

IDENT               	unit
=                   	=
INT                 	3
NEW_LINE            	

IDENT               	open
(                   	(
IDENT               	unit
=                   	=
IDENT               	unit
,                   	,
IDENT               	file
=                   	=
STRING              	'data.dat'
,                   	,
IDENT               	status
=                   	=
STRING              	'old'
,                   	,
IDENT               	action
=                   	=
STRING              	'read'
,                   	,
NEW_LINE            	

IDENT               	access
=                   	=
STRING              	'sequential'
,                   	,
IDENT               	form
=                   	=
STRING              	'formatted'
,                   	,
IDENT               	err
=                   	=
INT                 	910
)                   	)
NEW_LINE            	

IDENT               	read
(                   	(
IDENT               	unit
=                   	=
IDENT               	unit
,                   	,
IDENT               	fmt
=                   	=
STRING              	'(i10)'
,                   	,
IDENT               	advance
=                   	=
STRING              	'no'
,                   	,
IDENT               	size
=                   	=
IDENT               	inputsize
,                   	,
NEW_LINE            	

IDENT               	err
=                   	=
INT                 	920
)                   	)
IDENT               	val
NEW_LINE            	

IDENT               	inquire
(                   	(
IDENT               	unit
=                   	=
IDENT               	unit
,                   	,
IDENT               	recl
=                   	=
IDENT               	i1
,                   	,
IDENT               	access
=                   	=
IDENT               	c1
,                   	,
IDENT               	position
=                   	=
IDENT               	c2
)                   	)
NEW_LINE            	

IDENT               	write
(                   	(
IDENT               	unit
=                   	=
*                   	*
,                   	,
IDENT               	fmt
=                   	=
STRING              	'(1x,a7)'
,                   	,
IDENT               	advance
=                   	=
STRING              	'yes'
)                   	)
STRING              	'Success'
NEW_LINE            	

IDENT               	close
(                   	(
IDENT               	unit
)                   	)
NEW_LINE            	

return              	return
NEW_LINE            	

INT                 	910
IDENT               	print
*                   	*
,                   	,
STRING              	'Error opening file.'
NEW_LINE            	

return              	return
NEW_LINE            	

INT                 	920
IDENT               	print
*                   	*
,                   	,
STRING              	'Error reading file.'
NEW_LINE            	

return              	return
NEW_LINE            	

END                 	end
NEW_LINE            	

